### ISSUE TITLE
Reader compile-time derivation fails for some nested case classes

### ISSUE THREAD

**Issue by migesok** [andr83/scalaconfig#1]
Example:

``` scala
object Test extends App {
  case class User(name: String, password: String)
  case class Settings(user: User)

  import com.github.andr83.scalaconfig._

  println(ConfigFactory.load().as[Settings])
}
```

Compilation error:

> Error:(15, 34) diverging implicit expansion for type com.github.andr83.scalaconfig.HReader[this.Out]
> starting with method hconsHReader1 in object HReader
>   println(ConfigFactory.load().as[Settings])
> 
> Error:(15, 34) not enough arguments for method as: (implicit evidence$1: com.github.andr83.scalaconfig.Reader[com.evolutiongaming.onewallet.betting.integrations.evostd.Test.Settings])com.evolutiongaming.onewallet.betting.integrations.evostd.Test.Settings.
> Unspecified value parameter evidence$1.
>   println(ConfigFactory.load().as[Settings])

**Comment by andr83** [comment]
Thank you for reporting. The fix was published to snapshot. Can you check please?

**Comment by andr83** [comment]
Fixed in v0.2
